astropy/coordinates/baseframe.py

```diff
@@ -1239,16 +1239,18 @@ def __getattr__(self, attr):
         return self.__getattribute__(attr)  # Raise AttributeError.
 
     def __setattr__(self, attr, value):
-        repr_attr_names = set()
-        if hasattr(self, 'representation_info'):
-            for representation_attr in self.representation_info.values():
-                repr_attr_names.update(representation_attr['names'])
-
-        if attr in repr_attr_names:
-            raise AttributeError(
-                'Cannot set any frame attribute {0}'.format(attr))
-        else:
-            super().__setattr__(attr, value)
+        # Don't slow down access of private attributes!
+        if not attr.startswith('_'):
+            if hasattr(self, 'representation_info'):
+                repr_attr_names = set()
+                for representation_attr in self.representation_info.values():
+                    repr_attr_names.update(representation_attr['names'])
+
+                if attr in repr_attr_names:
```

Comment:
The logic has changed in that this block is now inside the above `if hasattr(..)` block.  Are you sure this is correct?

Reply:
Yes, since we know that without that block, `repr_attr_names` is an empty set.